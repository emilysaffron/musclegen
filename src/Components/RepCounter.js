import styled from "@emotion/styled";
import { useState } from "react";
import UpdateReps from "../Helpers/UpdateReps";
const StyledReps = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 5px;
  margin: 5px;
  align-items: center;
  border-radius: 20px;
`;

const StyledButton = styled.button`
  border-radius: 20px;
  background: #a8bbce;
  width: 150px;
  height: 150px;
  padding: 20px;
  margin: 10px;
  font-family: Palatino, URW Palladio L, serif;
  color: black;
  &:hover {
    cursor: pointer;
  }
  outline: none;
`;

const RepCounter = () => {
  const [repNumber, updateRepNumber] = useState(0);

  const handleClick = (operation) => {
    UpdateReps(updateRepNumber, repNumber, operation);
  };

  return (
    <StyledReps>
      <StyledButton onClick={() => handleClick("plus")}> + </StyledButton>
      <StyledButton onClick={() => handleClick("minus")}> - </StyledButton>
      {repNumber}
    </StyledReps>
  );
};

export default RepCounter;
