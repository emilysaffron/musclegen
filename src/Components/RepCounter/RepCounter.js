import styled from "@emotion/styled";

const StyledReps = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 5px;
  margin: 5px;
  align-items: center;
  border-radius: 20px;
`;

const StyledButton = styled.button`
  border-radius: 20px;
  background: #a8bbce;
  width: 150px;
  height: 90px;
  padding: 20px;
  margin: 10px;
  cursor: pointer;
  outline: none;
`;

const ButtonsContainer = styled.div`
  display: flex;
`;

const StyledNum = styled.div`
  background: #cea8bb;
  display: flex;
  flex-direction: column;
  flex-wrap: wrap;
  padding: 5px;
  color: black;
  align-items: center;
  border-radius: 50%;
  width: 40px;
  height: 40px;
  justify-content: center;
`;

const RepCounter = ({ repNumber, updateRepNumber }) => {
  const handleClick = (operation) => {
    if (operation === "plus") {
      updateRepNumber(repNumber + 1);
    }
    if (operation === "minus") {
      if (repNumber > 0) {
        updateRepNumber(repNumber - 1);
      }
    }
  };

  return (
    <StyledReps>
      <ButtonsContainer>
        <StyledButton onClick={() => handleClick("plus")}>
          Increase Reps
        </StyledButton>
        <StyledButton onClick={() => handleClick("minus")}>
          Decrease Reps
        </StyledButton>
      </ButtonsContainer>
      <StyledNum>{repNumber}</StyledNum>
    </StyledReps>
  );
};

export default RepCounter;
