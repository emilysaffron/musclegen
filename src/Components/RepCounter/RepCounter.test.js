import renderer from "react-test-renderer";
import RepCounter from "./RepCounter";

it("RepCounter Snapshot", () => {
  const tree = renderer
    .create(<RepCounter repNumber={1} updateRepNumber="updateRepNumber" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
