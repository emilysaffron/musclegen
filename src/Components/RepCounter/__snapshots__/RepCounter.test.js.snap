// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepCounter Snapshot 1`] = `
<div
  className="css-k46vyb"
>
  <div
    className="css-ov1ktg"
  >
    <button
      className="css-ikh6w2"
      onClick={[Function]}
    >
      Increase Reps
    </button>
    <button
      className="css-ikh6w2"
      onClick={[Function]}
    >
      Decrease Reps
    </button>
  </div>
  <div
    className="css-nu9ioa"
  >
    1
  </div>
</div>
`;
